libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp

```diff
@@ -43,11 +53,21 @@ void RCInput_ZYNQ::_timer_tick()
     uint32_t v;
 
     // all F's means no samples available
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ALTERA
+    while( (v = *((uint32_t*)(pulse_input + CUSTOM_PWM_0_BASE1)) ) != 0xffffffff ) {
```

Comment:
again... if you define `CUSTOM_PWM_0_BASE1` as 0 for the other case, you don't need the ifdef here.